pkg/query-service/rules/thresholdRule.go

```diff
@@ -447,6 +449,7 @@ func (r *ThresholdRule) runChQuery(ctx context.Context, db clickhouse.Conn, quer
 		if err := rows.Scan(vars...); err != nil {
 			return nil, err
 		}
+		r.lastTimestampWithDatapoints = time.Now()
```

Comment:
what if the QS restarts? If the new eval period does not have any instance of the fingerprint, `rows` will be empty. Should we use CH query to get the `lastTimestampWithDatapoints`?

Reply:
It would still trigger an alert because the default timestamp would be before but the `lastSeen` label we added below would be incorrect. I don't think there is a need to query the ClickHouse again. If it didn't fire when the query service was running but becomes true for new eval after the restart we know that past period data existed and can use that information for lastSeen.